src/csharp/Grpc.Core/Server.cs

```diff
@@ -325,25 +344,41 @@ private void DisposeHandle()
             {
```

Comment:
Well, not really. When an exception is thrown, HandleCallAsync eats the exception in the catch block and continues normally. So what I'm currently doing is safe I believe.
On the other hand, you're right that adding the finally block is probably more readable and maintainable for the future, so I will change that in a followup PR.